tests/assignments.test.ts

```diff
@@ -1,39 +1,372 @@
 import { Document } from './types/Document';
 import { last } from './utils/Arrays';
-import { getContent, getContentGroupByLabel, getDetailsGroup, getDocumentsByType } from './utils/Dao';
-import { nextDay, setDate } from './utils/Date';
+import { getContent, getContentGroupByLabel, getDetailsGroup, getDocumentByHash, getDocumentsByType, getEdgesByFilter } from './utils/Dao';
 import { getDaoExpect } from './utils/Expect';
 import { UnderwaterBasketweaver } from './sample-data/RoleSamples';
 import { DaoBlockchain } from './dao/DaoBlockchain';
 import { getAssignmentProposal } from './sample-data/AssignmentSamples';
-import { proposeAndPass } from './utils/Proposal';
+import { passProposal, proposeAndPass } from './utils/Proposal';
+import { assetToNumber } from './utils/Parsers';
+import { PHASE_TO_YEAR_RATIO } from './utils/Constants';
+import { DocumentBuilder } from './utils/DocumentBuilder';
+import { setDate } from './utils/Date';
+import { getAccountPermission } from './utils/Permissions';
+
+const getStartPeriod = (environment: DaoBlockchain, assignment: Document): Document => {
+
+    const assignmentDetails = getContentGroupByLabel(assignment, 'details');
+
+    const startPeriod = getContent(assignmentDetails, 'start_period').value[1];
+
+    return getDocumentByHash(environment.getDaoDocuments(), startPeriod as string);
+}
+
+const getPeriodStartDate = (period: Document): Date => {
+
+    const details = getContentGroupByLabel(period, 'details');
+
+    const startDate = getContent(details, 'start_time');
+
+    return new Date(startDate.value[1]);
+}
+
+const getEditProposal = (original: string,
+                         newTitle: string, 
+                         newPeriods: number): Document => {
+    return DocumentBuilder
+    .builder()
+    .contentGroup(builder => {
+      builder
+      .groupLabel('details')
+      .string('title', newTitle)
+      .string('ballot_description', 'Edit assignment')
+      .checksum256('original_document', original)
+      .int64('period_count', newPeriods)
+    })
+    .build();
+}
+
+const claimRemainingPeriods = async (assignment: Document,
+                                     environment: DaoBlockchain): Promise<number> => {
+
+    let claimedPeriods = 0;
+    let edges = environment.getDaoEdges();
+    let docs = environment.getDaoDocuments();
+
+    let currentPeriod = getStartPeriod(environment, assignment);
+
+    while (true) {
+
+        let nextEdge = getEdgesByFilter(edges, { from_node: currentPeriod.hash, edge_name: 'next' });
+
+        expect(nextEdge).toHaveLength(1);
+
+        const nextPeriod = getDocumentByHash(docs, nextEdge[0].to_node);
+
+        let now = getPeriodStartDate(nextPeriod);
+
+        setDate(environment, now, 0);
+
+        try {
+          await environment.dao.contract.claimnextper({
+            assignment_hash: assignment.hash
+          });
+        }
+        catch(error) {
+          console.log("Error while claiming period:", error)
```

Comment:
shouldn't this error fail the test?

Reply:
Not really since the intent for this helper function is to claim all the remaining periods, which could be i.e. 0 and in that scenario it will fail the transaction with the error 'all available periods were claimed' and return 0. Maybe the intent is not quite clear here so I'll refactor it. Anyways, I should check that the error string is the one that I mentioned and also remove the log.